store/src/java/com/zimbra/cs/store/StoreManager.java

```diff
@@ -68,18 +92,101 @@ public static StoreManager getInstance(String className) {
         return sInstance;
     }
 
-
+    /**
+     *  Reset StoreManager for current volume
+     *  TODO: Add documentation here
+     * @throws ServiceException
+     * @throws IOException
+     */
+    public static void resetStoreManager() throws ServiceException, IOException {
+        if (Provisioning.getInstance().getLocalServer().isSMRuntimeSwitchEnabled()) {
+            ZimbraLog.store.error("isSMRunTimeSwitchEnabled is not enabled");
+        }
+        if(ImapDaemon.isRunningImapInsideMailboxd()) {
+            synchronized (StoreManager.class) {
+                ZimbraLog.store.error("resetting StoreManager: Thread: %s", Thread.currentThread().getName());
```

Comment:
Is `error` level really required, can't use info or debug?

Reply:
it was a mistake. corrected